src/opnsense/mvc/app/library/OPNsense/Backup/Mailer.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace OPNsense\Backup;
+
+require_once('PHPMailer/SMTP.php');
+require_once('PHPMailer/Exception.php');
+require_once('PHPMailer/PHPMailer.php');
+
+use OPNsense\MailSender;
+use OPNsense\Core\Config;
+use PHPMailer\PHPMailer\SMTP;
+use PHPMailer\PHPMailer\Exception;
+use PHPMailer\PHPMailer\PHPMailer;
+
+/**
+ * Class mail backup
+ * @package OPNsense\Backup
+ */
+class Mailer extends Base implements IBackupProvider
+{
+
+    /**
+     * get required (user interface) fields for backup connector
+     * @return array configuration fields, types and description
+     */
+    public function getConfigurationFields()
+    {
+        $fields = array();
+
+        $fields[] = array(
+            "name" => "MailEnabled",
+            "type" => "checkbox",
+            "label" => gettext("Enable"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "Receiver",
+            "type" => "text",
+            "label" => gettext("Receiver Email Address"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpHost",
+            "type" => "text",
+            "label" => gettext("SMTP Host"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpPort",
+            "type" => "text",
+            "label" => gettext("SMTP Port"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpSSL",
+            "type" => "checkbox",
+            "label" => gettext("Use SSL"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpUsername",
+            "type" => "text",
+            "label" => gettext("SMTP Username (Optional)"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "SmtpPassword",
+            "type" => "password",
+            "label" => gettext("SMTP Password (Optional)"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "GpgEmail",
+            "type" => "text",
+            "label" => gettext("GPG Email"),
+            "value" => null
+        );
+        $fields[] = array(
+            "name" => "GpgPublicKey",
+            "type" => "textarea",
+            "label" => gettext("GPG Public Key"),
+            "value" => null
+        );
```

Comment:
As an alternative, you should think about S/MIME since it does not require additional software in common mail clients.
OpenSSL / LibreSSL can do it for you if you don't want to do it manually: `man openssl-smime`